ldb/tx.go

```diff
@@ -346,3 +375,203 @@ func (db *LevelDb) FetchTxBySha(txsha *btcwire.ShaHash) ([]*btcdb.TxListReply, e
 	}
 	return replies, nil
 }
+
+// addrIndexToKey serializes the passed txAddrIndex for storage within the DB.
+// Transactions indexed by address are stored with the following format:
+//   * prefix || hash160 || blockHeight || txoffset || txlen
+// Indexes are stored purely in the key, with blank data for the actual value
+// in order to facilitate ease of iteration by their shared prefix and
+// also to allow limiting the number of returned transactions (RPC).
+// Alternatively, indexes for each address could be stored as an
+// append-only list for the stored value. However, this add unnecessary
+// overhead when storing and retrieving since the *entire* list must
+// be fetched each time.
+func addrIndexToKey(index *txAddrIndex) []byte {
+	record := make([]byte, addrIndexKeyLength, addrIndexKeyLength)
+	copy(record[:2], addrIndexKeyPrefix)
+	copy(record[2:22], index.hash160[:])
+
+	// The index itself.
+	binary.LittleEndian.PutUint64(record[22:30], uint64(index.blkHeight))
+	binary.LittleEndian.PutUint32(record[30:34], uint32(index.txoffset))
```

Comment:
Is 4 bytes enough for the offset?

Put another way, does this already break the in-memory representation on 32-bit due to int being 4 bytes there?